proteobench/modules/quant/benchmarking.py

```diff
@@ -0,0 +1,253 @@
+
+
+def handle_benchmarking_error(error_type: Type[Exception], error_message: str):
+    """
+    Decorator to handle benchmarking errors with custom error messages.
+
+    Parameters
+    ----------
+    error_type : Type[Exception]
+        The type of exception to catch
+    error_message : str
+        The error message to raise if the exception occurs
+    """
+
+    def decorator(func: Callable):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except error_type as e:
+                raise error_type(f"{error_message}: {e}")
+            except Exception as e:
+                raise error_type(f"Unexpected error in {func.__name__}: {e}")
+
+        return wrapper
+
+    return decorator
+
+
+@handle_benchmarking_error(ParseError, "Error parsing input file")
+def _load_input(input_file: str, input_format: str) -> DataFrame:
+    """Load and parse the input file."""
+    return load_input_file(input_file, input_format)
+
+
+@handle_benchmarking_error(ParseSettingsError, "Error parsing settings")
+def _load_settings(parse_settings_dir: str, module_id: str, input_format: str):
+    """Load and parse the settings file."""
+    return ParseSettingsBuilder(parse_settings_dir=parse_settings_dir, module_id=module_id).build_parser(input_format)
+
+
+@handle_benchmarking_error(ConvertStandardFormatError, "Error converting to standard format")
+def _convert_format(parse_settings, input_df: DataFrame):
+    """Convert input to standard format."""
+    return parse_settings.convert_to_standard_format(input_df)
+
+
+@handle_benchmarking_error(QuantificationError, "Error generating quantification scores")
+def _create_quant_scores(precursor_column_name: str, parse_settings):
+    """Create quantification scores."""
+    return QuantScores(precursor_column_name, parse_settings.species_expected_ratio(), parse_settings.species_dict())
+
+
+@handle_benchmarking_error(IntermediateFormatGenerationError, "Error generating intermediate data structure")
+def _generate_intermediate(quant_score, standard_format, replicate_to_raw):
+    """Generate intermediate data structure."""
+    return quant_score.generate_intermediate(standard_format, replicate_to_raw)
+
+
+@handle_benchmarking_error(DatapointGenerationError, "Error generating datapoint")
+def _generate_datapoint(intermediate_metric_structure, input_format, user_input, default_cutoff_min_prec):
+    """Generate datapoint."""
+    return QuantDatapoint.generate_datapoint(
+        intermediate_metric_structure, input_format, user_input, default_cutoff_min_prec=default_cutoff_min_prec
+    )
+
+
+@handle_benchmarking_error(DatapointAppendError, "Error adding current data point")
+def _append_datapoint(add_datapoint_func, current_datapoint, all_datapoints):
+    """Append datapoint to all datapoints."""
+    return add_datapoint_func(current_datapoint, all_datapoints=all_datapoints)
+
+
+def run_benchmarking(
```

Comment:
Is this one different from the one in `webinterface/pages/base_pages/quant.py` ? I guess it is aggregating some of the functions there?